### DISCUSSION TITLE
Outlook Safelink invalidates login token

### DISCUSSION THREAD

**Discussion by pi43r** [supabase/supabase#28903] [Questions] [answered]
Hi, 

I followed this [guide](https://supabase.com/docs/guides/auth/sessions/pkce-flow) using nextjs and server side rendering to log users in. I had a complaint telling me that their reset password link did not work and it turns out that outlook wraps my URL into a safelink that look like this `https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwebsite.com%2Fauth%2Fconfirm%3Ftoken_hash%3D38b74f...`

And it seems that the token is invalidated during the redirect so my user will get to my error page.

How to mitigate this?

**Comment by GaryAustin1** [comment, marked as answer]
Not sure this helps, but it is the documented work around for this type of issue:  https://supabase.com/docs/guides/auth/auth-email-templates#email-prefetching

**Comment by pi43r** [comment]
Thank you, I guess I missed that in the docs.
I think that's a reasonable workaround
